examples/device/ep/nixl_ep/buffer.py

```
@@ -81,11 +82,18 @@ def __init__(
                 In low-latency paths, a timeout marks the rank invalid and masks it out.
                 In high-throughput paths, a timeout is fatal and traps.
                 Default: 30000 ms.
+            nvl_group_size: number of ranks in one CUDA-IPC/NVLink-local group for high-throughput EP.
+                Defaults to 8 for backward compatibility. GB200 deployments that expose 4 GPUs per
+                worker should pass 4 so CUDA IPC is only opened within the local worker group.
         """
+        assert 0 < nvl_group_size <= 8 and 8 % nvl_group_size == 0, (
+            "nvl_group_size must divide the fixed 8-lane NVL scratch layout"
+        )
```

Ruff (0.15.17)

[warning] Assertion should be broken down into multiple parts

(PT018)